harmonica/forward/prisms_layer.py

```diff
@@ -0,0 +1,411 @@
+        Computes the gravity generated by the layer of prisms
+
+        Uses :func:`harmonica.prism_gravity` for computing the gravity field
+        generated by the prisms of the layer.
+        The density of the prisms will be assigned from the ``data_var`` chosen
+        through the ``density_name`` argument.
+        Ignores the prisms which ``top`` or ``bottom`` boundaries are
+        ``np.nan``s.
+        All ``kwargs`` will be passed to :func:`harmonica.prism_gravity`.
+
+        Parameters
+        ----------
+        coordinates : list or 1d-array
+            List or array containing ``easting``, ``northing`` and ``upward``
+            of the computation points defined on a Cartesian coordinate system.
+            All coordinates should be in meters.
+        field : str
+            Gravitational field that wants to be computed.
+            The available fields are:
+            - Gravitational potential: ``potential``
+            - Downward acceleration: ``g_z``
+        density_name : str (optional)
+            Name of the property layer (or ``data_var`` of the
+            :class:`xarray.Dataset`) that will be used for the density of each
+            prism in the layer. Default to ``"density"``
+
+        Returns
+        -------
+        result : array
+            Gravitational field generated by the prisms on the computation
+            points in mGal.
+
+        See also
+        --------
+        harmonica.prism_gravity
+        """
+        boundaries = self.get_prisms()
+        # Get the mask for selecting only the boundaries of the prisms that
+        # have no nans on their top and bottom
+        nonans_boundaries = self._get_nonans_prisms_mask().ravel()
+        # Get density array
+        density = getattr(self._obj, density_name).values.ravel()
+        # Select only the boundaries and density elements for nonans prisms
+        boundaries = boundaries[nonans_boundaries]
+        density = density[nonans_boundaries]
+        # Check if the new density array has no nans
+        if np.isnan(density).any():
+            raise ValueError(
+                "Invalid density property found: "
+                + "the density array shouldn't contain any np.nan's."
+            )
+        return prism_gravity(
+            coordinates,
+            prisms=boundaries,
+            density=density,
+            field=field,
+            **kwargs,
+        )
+
+    def _get_nonans_prisms_mask(self):
+        """
+        Build a mask for selecting prisms with no nans on top and bottom
+
+        Returns
+        -------
+        mask : 2d-array
+            Array of bools that can be used as a mask for selecting prisms with
+            no nans on top and bottom boundaries.
+        """
+        nonans_top = np.logical_not(np.isnan(self._obj.top.values))
+        nonans_bottom = np.logical_not(np.isnan(self._obj.bottom.values))
+        return np.logical_and(nonans_top, nonans_bottom)
+
+    def get_prisms(self):
```

Comment:
```suggestion
    def to_prisms(self):
```

This fits better with the rest of the xarray methods: http://xarray.pydata.org/en/stable/pandas.html

Reply:
Could keep this private for now and provide it after if desired.